WordPress/gutenberg
# Mobile - Preformatted block - Fix showing the default background color

Fixes https://github.com/WordPress/gutenberg/issues/36874

- Gutenberg Mobile PR -> https://github.com/wordpress-mobile/gutenberg-mobile/pull/4292

## Description
This PR fixes a regression introduced after these changes https://github.com/WordPress/gutenberg/pull/35314. Before it was checking for `style?.baseColors` but when there's no block-based theme active, it returns an empty object. So it was affecting standard themes.

Now it'll check the object exists and it's not empty.

## How has this been tested?
### Test case 1 - Standard theme

- Open the app
- Add a `Preformatted` block.
- **Expect** to see the default background, it should work for both light/ and dark mode.

### Test case 2 - Standard theme (custom colors)

- Open the app
- Add a `Preformatted` block.
- **Expect** to see the default background
- Open the block settings
- Change the background color
- **Expect** to see the custom color in the block

### Test case 3 - Block-based theme

- Open the app
- Add a `Preformatted` block.
- **Expect** to **NOT** see the default background, current block-based themes don't apply a background color for this block.
- 
### Test case 4 - Block-based theme (custom colors)

- Open the app
- Add a `Preformatted` block.
- **Expect** to **NOT** see the default background, current block-based themes don't apply a background color for this block.
- Open the block settings
- Change the background color
- **Expect** to see the custom color in the block

## Screenshots 

Test case 1 (Light) | Test case 2 (Dark) | Test case 3 (Custom color)
-|-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/143485908-e8dfaf3c-d8f8-46a6-bb40-35e2c3c319a4.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/143485909-fbd50c04-9698-4b20-ab3d-2aced7c76c5a.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/143485920-2aa6bb39-4b81-4174-bd35-11e64b3c43b6.png" width="200" /></kbd> 

Test case 3 | Test case 4 (Custom color)
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/143485988-191cae63-182a-42f0-bb2c-45bb51091f68.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/143485994-6015f4cc-56d1-43f9-9bc6-37720c33c30e.png" width="200" /></kbd>

## Types of changes
Bug fix

## Checklist:
